### PR TITLE
Add multipleResolves event support

### PR DESCRIPTION
**Pull request by mcollina**
Fixes https://github.com/mcollina/make-promises-safe/issues/7.

@BridgeAR @targos @jasnell @benjamingr can you please review this?

### CODE REVIEW

**Comment by mcollina** [comment]
(this is a semver-major change)

File: make-promises-safe.js [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
-    process.exit(1)
-  })
+install('unhandledRejection', function (err) {
+  console.error(err)
+  abortOrExit()
+})
+
+install('multipleResolves', function (type, promise, reason) {
+  console.error(type, promise, reason)
+  abortOrExit()
```
Comment (benjamingr) [inline make-promises-safe.js]: I'm not sure if it's actually "safe" to do this based on actual real world usage - but sure :) (the error - for sure, the exit - not 100% sure)

File: README.md [hunk 2]
```diff
@@ -79,25 +95,73 @@ function doStuff () {
...
+and exits the process with an exit code of 1.
+
+This feature is available only on Node >= 10.12.0.
+
 ## License
 
 MIT
 
 [unhandled]: https://nodejs.org/dist/latest-v8.x/docs/api/deprecations.html#deprecations_dep0018_unhandled_promise_rejections
+[unhandled-event]: https://nodejs.org/dist/latest-v10.x/docs/api/process.html#process_event_unhandledrejection)
```
Comment (targos) [inline README.md]: extra parenthesis at the end

File: README.md [hunk 3]
```diff
@@ -79,25 +95,73 @@ function doStuff () {
...
 ```
-require('make-promises-safe').abort = true
-```
+
+Under load, the above script will exits with an EMFILE error that would
+be extremely hard to debug.
+
+### The Solution
+
+`make-promises-safe` installs an [`process.on('multipleResolves')`][multiple-event]
```
Comment (targos) [inline README.md]: there's no "multiple-event" link

**Pull request merged**